### PR TITLE
Update the README to be much more informative and up-to-date.

### PR DESCRIPTION
**Pull request by Cifram**
Add a lot of detail to the README.md.

* Now breaks up the raw API description with basic information about the different noise functions.
* Updates the Brownian API to reflect what's actually in place.
* Updates the worley noise API to be called cell noise, and reflect my current thinking about how it should be implemented.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,58 +9,201 @@ Procedural noise generation library for Rust.
...
+Perlin noise is a very fast and reasonable quality gradient noise.
+~~~rust
 fn perlin2<T: Float>(seed: &Seed, point: &Point2<T>) -> T;
 fn perlin3<T: Float>(seed: &Seed, point: &Point3<T>) -> T;
 fn perlin4<T: Float>(seed: &Seed, point: &Point4<T>) -> T;
+~~~
 
-fn simplex2<T: Float>(seed: &Seed, point: &Point2<T>) -> T;
-fn simplex3<T: Float>(seed: &Seed, point: &Point3<T>) -> T;
+OpenSimplex noise is a slower but higher quality form of graidnet noise.
```
Comment (amaranth) [inline README.md]: Misspelled gradient.

File: README.md [hunk 2]
```diff
@@ -9,58 +9,201 @@ Procedural noise generation library for Rust.
...
-fn worley2_manhattan_point<T: Float>(seed: &Seed, point: &Point2<T>) -> T;
-fn worley3_manhattan_point<T: Float>(seed: &Seed, point: &Point3<T>) -> T;
-fn worley4_manhattan_point<T: Float>(seed: &Seed, point: &Point4<T>) -> T;
+These are the set of functions for determining range between two points, which are used to determine which seed point is actually closest to the test point.
 
-fn worley2_manhattan_edge<T: Float>(seed: &Seed, point: &Point2<T>) -> T;
-fn worley3_manhattan_edge<T: Float>(seed: &Seed, point: &Point3<T>) -> T;
-fn worley4_manhattan_edge<T: Float>(seed: &Seed, point: &Point4<T>) -> T;
+Calculate the euclidian range between two points. This is what one normally things of as distance.
+~~~rust
```
Comment (amaranth) [inline README.md]: Should these be public?

File: README.md [hunk 3]
```diff
@@ -9,58 +9,201 @@ Procedural noise generation library for Rust.
...
+fn range_euclidian2<T: Float>(p1: Point2<T>, p2: Point2<T>) -> T;
+fn range_euclidian3<T: Float>(p1: Point3<T>, p2: Point3<T>) -> T;
+fn range_euclidian4<T: Float>(p1: Point4<T>, p2: Point4<T>) -> T;
+~~~
+
+Calculate the manhattan range between two points. This is the sum of the coordinates in the offset vector, equivalent to the distance one would walk on the Manhattan streets between two points.
+~~~rust
```
Comment (amaranth) [inline README.md]: Same with these.

**Comment by Cifram** [comment]
Fixed the typo, and explained the need for the various cell noise functions to be public in IRC.

To re-iterate, the range of different kinds of cell noise is immense. We can't feasibly implement them all. The point generation functions, though, are the common logic they'll all need. By exposing them, we make it possible for users of this library to implement any kind of cell noise we don't have. The range functions need to be exposed because they're required to call the point generation functions, and having the range function injected means the library's users can also add their own range functions.

**Pull request merged**